pkg/storage/unified/resource/sqlkv.go

```
@@ -114,6 +116,32 @@ func (req sqlKVDeleteRequest) Validate() error {
 	return req.sqlKVSectionKey.Validate()
 }
 
+type sqlKVKeysRequest struct {
+	sqltemplate.SQLTemplate
+	sqlKVSection
+	Options ListOptions
+}
+
+func (req sqlKVKeysRequest) Validate() error {
+	return req.sqlKVSection.Validate()
+}
+
+func (req sqlKVKeysRequest) StartKey() string {
+	return req.Section + "/" + req.Options.StartKey
+}
+
+func (req sqlKVKeysRequest) EndKey() string {
+	if req.Options.EndKey == "" {
+		req.Options.EndKey = PrefixRangeEnd(req.Section + "/")
+	}
+
+	return req.Section + "/" + req.Options.EndKey
+}
+
+func (req sqlKVKeysRequest) SortAscending() bool {
```

the default should be to sort ascending, it seems like the way you're doing it, it defaults to descending yeah?

---

Unless you are talking about the situation when a caller explicitly uses an invalid `SortOrder` (like `ListOptions{Sort: SortOrder(42)}`), in which case, yes, it will use DESC order (sigh, no enums). I would argue we should return an error in that case instead of having a default.